First part of the part-two translation (#3)
horacioh/gatsby: docs/tutorial/part-eight/index.md

---
title: Preparar el sitio web para montarlo en línea.
typora-copy-images-to: ./
disableTableOfContents: true
---

Wow! ¡Haz recorrido un largo camino! Haz aprendido a:

- crear nuevos sitios web con Gatsby
- crear páginas y componentes

> - darle estilo a componentes

Reviewer: quizas mejor `estilar componentes` ?

Author: te refieres a 
```suggestion
- estilizar componentes
```
?

Reviewer: sip!

- agregar plugins a un sitio web
- agregar y transformar datos
- usar GraphQL para hacer consultas de datos para tus páginas web
- crear páginas web programáticamente utilizando tus datos

> En esta sección final, daremos algunos pasos comunes para preparar un sitio web y montarlo en línea, utilizando una poderosa herramienta que diagnostica sitios web llamada [Lighthouse](https://developers.google.com/web/tools/lighthouse/). A lo largo del camino introduciremos algunos plugins que comúnmente utilizarás en tus sitios web Gatsby.

Reviewer: `En esta sección final, te guiaremos en algunos pasos básicos para preparar para publicar tu página web, utilizando una potente herramienta de diagnóstico web llamada Lighthouse.
`

qué te parece?

Author: ```suggestion
En esta sección final, te guiaremos en algunos pasos básicos para preparar y publicar tu página web, utilizando una potente herramienta de diagnóstico web llamada [Lighthouse](https://developers.google.com/web/tools/lighthouse/). A lo largo del camino introduciremos algunos plugins que comúnmente utilizarás en tus sitios web Gatsby.
```
solo cambié un "para" por un "y"

Reviewer: 👍

## Auditoría con Lighthouse

> Citando del [sitio web Lighthouse](https://developers.google.com/web/tools/lighthouse/):

Reviewer: que tal "Textualmente"? o "Citando Textualmente"?

Author: ```suggestion
Citando textualmente de [Lighthouse](https://developers.google.com/web/tools/lighthouse/):
```

> Lighthouse es una herramienta automatizada de código abierto para aumentar la calidad de las páginas web. Puedes ejecutarla para que diagnostique cualquier pagina web, publica o que requiera autenticación. Audita el desempeño, accesibilidad, aplicación web progresiva (PWAs siglas en ingles), y más.

> Lighthouse viene incluido en Chrome DevTools. Ejecutar sus auditorías -- y después abordando los errores que encuentra e implementando las mejoras que sugiere -- es una manera estupenda de preparar tu sitio web para ponerlo en línea. Su ayuda te da la confianza de que tu sitio web es lo mas rápido y accesible posible.

Reviewer: "Ejecutando sus auditorías..."

Author: ```suggestion
Lighthouse viene incluido en Chrome DevTools. Ejecutando sus auditorías -- y después abordando los errores que encuentra e implementando las mejoras que sugiere -- es una manera estupenda de preparar tu sitio web para ponerlo en línea. Su ayuda te da la confianza de que tu sitio web es lo mas rápido y accesible posible.
```

¡Probémoslo!

> Primero, necesitas generar un compilado para producción de tu sitio web Gatsby. El servidor de desarrollo de Gatsby está optimizado para hacer el desarrollo rápido; pero el sitio web que genera, aunque se acerca a una versión de producción, no está igual de optimizado.

Reviewer: "Primero, necesitas generar un compilado para producción de tu sitio web Gatsby. El servidor de desarrollo de Gatsby está optimizado para hacer el proceso de desarrollo rápido;"

Author: ```suggestion
Primero, necesitas generar un compilado para producción de tu sitio web Gatsby. El servidor de desarrollo de Gatsby está optimizado para hacer rápido el proceso de desarrollo; pero el sitio web que genera, aunque se acerca a una versión de producción, no está igual de optimizado.
```

### ✋ Genera un compilado para producción

1.  Detén el servidor de desarrollo (si es que sigue en ejecución) y ejecuta el siguiente comando:

```shell
gatsby build
```

> > 💡 Como aprendiste en la [parte 1](/tutorial/part-one/), esto genera un compilado para producción de tu sitio web y crea los archivos estáticos en el directorio `public`.

Reviewer: quizás cambiamos el link a ´/tutorial/primera-parte´?

Author: es complicado saber los enlaces, ya que yo hubiera decidido "parte-uno", es por eso que los dejé creo que todos intactos, preferiría dejarlos así y al final que estén todos listos hacer los cambios a todos los archivos, sin embargo, si insistes hago el cambio de inmediato.

Reviewer: me parece bien dejarlos todos como estan ahora :)

2.  Observa el sitio web de producción localmente. Ejecuta:

```shell
gatsby serve
```

Una vez iniciado, puedes observar tu sitio web en [`localhost:9000`](http://localhost:9000).

### Ejecuta una auditoría Lighthouse

Ahora ejecutarás tus primeras pruebas con Lighthouse.

1.  Si aun no lo haz hecho, abre tu sitio web en Chrome en modo incógnito, para que las extensiones no interfieran con las pruebas. Después, abre la Chrome DevTools.

> 2.  Da clic en la pestaña "Audits" donde mirarás una pantalla como esta:

Reviewer: "Haz click en la pestaña..."

Author: ```suggestion
2.  Haz clic en la pestaña "Audits" donde mirarás una pantalla como esta:
```

![Lighthouse audit start](./lighthouse-audit.png)

3.  Da clic en "Perform an audit..." (todos los tipos de auditoría disponibles deberán estar seleccionadas por defecto). Después da clic en "Run audit" (tomará un minuto aproximadamente para hacer la auditoría). Una vez que la auditoría termine, verás unos resultados como estos:

![Lighthouse audit results](./lighthouse-audit-results.png)

> Como ves, el desempeño de Gatsby es excelente sin configuraciones previas, pero aún faltan algunas cosas para que sea PWA, accesibilidad, mejores practicas y SEO que mejorará tus resultados en búsquedas web (en el proceso haz tu sitio web mucho mas amigable para los visitantes y motores de búsqueda)

Reviewer: puedes revisar este párrafo?

Author: ```suggestion
Como puedes observar, el desempeño de Gatsby es excelente sin configuraciones previas, pero aún falta hacer algunas configuraciones para que el sitio web sea una PWA como accesibilidad, mejores practicas y SEO que mejorarán tus calificaciones (en el proceso haz tu sitio web mucho más amigable para los visitantes y motores de búsqueda).
```

Pero primero, ¿que _son_ exactamente las PWA?

Son sitios web normales que toman ventaja de la funcionalidad de los navegadores modernos para mejorar la experiencia de uso con opciones y beneficios como si se tratara de una aplicación. Mira el [enfoque de Google](https://developers.google.com/web/progressive-web-apps/) de lo que debe ser una experiencia PWA.

Incluir un manifiesto a la aplicación web es uno de los tres generalmente aceptados [requerimientos básicos para una PWA](https://alistapart.com/article/yes-that-web-project-should-be-a-pwa#section1).

Citando a [Google](https://developers.google.com/web/fundamentals/web-app-manifest/):

> El manifiesto de una aplicación web es un simple archivo JSON que le dice al navegador acerca de tu aplicación web y como comportarse cuando se 'instale' en el dispositivo movil o de escritorio.

El [plugin Gatsby's manifest](/packages/gatsby-plugin-manifest/) configura Gatsby para crear un archivo `manifest.webmanifest` en cada sitio web compilado.

### ✋ Usando `gatsby-plugin-manifest`

1.  Instala el plugin:

```shell
npm install --save gatsby-plugin-manifest
```

> 2. Agrega un favicon para tu aplicación en `src/images/icon.png`. Para este tutorial puedes usar este [icono de ejemplo](https://raw.githubusercontent.com/gatsbyjs/gatsby/master/docs/tutorial/part-eight/icon.png), no debes disponer de uno. El icono es necesario para crear todas las imágenes para el manifiesto. Para mas información, mira los docs [`gatsby-plugin-manifest`](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-plugin-manifest/README.md).

Reviewer: "El icono es necesario para crear todas las imágenes del manifiesto."

Author: ```suggestion
2. Agrega un favicon para tu aplicación en `src/images/icon.png`. Para este tutorial puedes usar este [icono de ejemplo](https://raw.githubusercontent.com/gatsbyjs/gatsby/master/docs/tutorial/part-eight/icon.png), no debes disponer de uno. El icono es necesario para crear todas las imágenes del manifiesto. Para más información, mira la documentación [`gatsby-plugin-manifest`](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-plugin-manifest/README.md).
```

        start_url: `/`,
        background_color: `#6b37bf`,
        theme_color: `#6b37bf`,

>         // Muestra mensaje "Agregar a escritorio" y deshabilita la UI del navegador (incluido el botón "atrás")

Reviewer: "Muestra el mensaje..."

Author: ```suggestion
        // Muestra el mensaje "Agregar a escritorio" y deshabilita la UI del navegador (incluido el botón "atrás")
```

        // ve https://developers.google.com/web/fundamentals/web-app-manifest/#display
        display: `standalone`,
        icon: `src/images/icon.png`, // Esta ruta es relativa a la raíz de el sitio web.
      },
    },
  ]
}
```

> Es todo lo que necesitas para iniciar a agregar un manifiesto a tu sitio web Gatsby. El ejemplo dado refleja una configuración base -- mira la [referencia de plugin](/packages/gatsby-plugin-manifest/?=gatsby-plugin-manifest#automatic-mode) para mas opciones.

Reviewer: "Ésto es todo lo que necesitas para empezar a agregar un..."

Author: ```suggestion
Es todo lo que necesitas para empezar a agregar un manifiesto a tu sitio web Gatsby. El ejemplo dado refleja una configuración base -- mira la [referencia de plugin](/packages/gatsby-plugin-manifest/?=gatsby-plugin-manifest#automatic-mode) para más opciones.
```

## Agrega soporte sin conexión

> Otro requerimiento para que un sitio web califique como PWA es el uso de un [service worker](https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API). un "service worker" se ejecuta en el fondo de atrás, decidiendo si ofrece el contenido en línea o lo almacenado en cache, basándose en la conectividad actual, permitiendo una agradable experiencia sin conexión.

Reviewer: Un "service worker" se ejecuta en segundo plano, decidiendo si ofrece recursos en linea o almacenados en la cache del navegador dependiendo del estado de conexión, permitiendo una experiencia agradable sin ella.

Author: ```suggestion
Otro requerimiento para que un sitio web califique como PWA es el uso de un [service worker](https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API). Un "service worker" se ejecuta en segundo plano, decidiendo si ofrece recursos en línea o almacenados en la cache del navegador dependiendo del estado de conexión, permitiendo una experiencia agradable sin ella.
```

El [plugin Gatsby's offline](/packages/gatsby-plugin-offline/) hace que el sitio web Gatsby funcione sin conexión y sea mas resistente a malas condiciones de red, creando un "service worker" para tu sitio web.

### ✋ Usando `gatsby-plugin-offline`

1.  Instala plugin:

```shell
npm install --save gatsby-plugin-offline
```

> 2.  Agrega el plugin al arreglo `plugins` en tu archivo `gatsby-config.js`.

Reviewer: Agrega el plugin al listado de `plugins` en el archivo `gatsby-config.js`.

Author: ```suggestion
2.  Agrega el plugin al listado de `plugins` en el archivo `gatsby-config.js`.
```

        start_url: `/`,
        background_color: `#6b37bf`,
        theme_color: `#6b37bf`,

>         // Muestra mensaje "Agregar a escritorio" y deshabilita la UI del navegador (incluido el botón "atrás")

Reviewer: "Muestra el mensaje..."

Author: ```suggestion
        // Muestra el mensaje "Agregar a escritorio" y deshabilita la UI del navegador (incluido el botón "atrás")
```

}
```

Es todo lo que necesitas para iniciar con "service workers" en Gatsby.

> > 💡 El plugin sin conexión (offline plugin) lo deberás listar _después_ del plugin manifiesto (manifest plugin), para que el plugin sin conexión pueda guardar en cache el archivo creado `manifest.webmanifest`.

Reviewer: "..., para que el plugin de conexión pueda guardar en la cache del navegador el archivo..."

Author: ```suggestion
> 💡 El plugin sin conexión (offline plugin) lo deberás listar _después_ del plugin manifiesto (manifest plugin), para que el plugin sin conexión pueda guardar en la cache del navegador el archivo creado `manifest.webmanifest`.
```

> ## Agregar metadata a la página

Reviewer: Agregar metadatos de la página

Author: ```suggestion
## Agregar metadatos a la página
```

> Agregar metadata a las páginas (como un título o descripción) es prioritario para ayudar a los motores de búsqueda como Google para entender tu contenido y decidir cuando mostrarte en sus resultados de búsqueda.

Reviewer: "Agregar metadata a las páginas (como un título o descripción) es clave para ayudar a motores de búsqueda como Google a entender..."

Author: ```suggestion
Agregar metadatos a las páginas (como un título o descripción) es clave para ayudar aa motores de búsqueda como Google a entender tu contenido y decidir cuando mostrarte en sus resultados de búsqueda.
```

> [React Helmet](https://github.com/nfl/react-helmet) es un paquete que provee a React componentes de interfaz para ti, para administrar tu [documento cabecera](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/head).

Reviewer: "[React Helmet](https://github.com/nfl/react-helmet) es una librería que provee una interfaz de componentes React para que administres la [cabecera de tus documentos HTML](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/head)."

Author: ```suggestion
[React Helmet](https://github.com/nfl/react-helmet) es una biblioteca que provee una interfaz de componentes React para que administres la [cabecera de tus documentos HTML](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/head).
```

> El plugin de Gatsby [react helmet](/packages/gatsby-plugin-react-helmet/) provee soporte a servidores, creando representaciones de datos agregados con React Helmet. Usando el plugin, los atributos agregados a React Helmet serán agregados a las páginas HTML que Gatsby compile.

Reviewer: "El plugin gatsby-plugin-react-helmet soporta renderizado en el servidor de toda información agregara con React Helmet."

Author: ```suggestion
El plugin [gatsby-plugin-react-helmet](/packages/gatsby-plugin-react-helmet/) soporta renderizado en el servidor de toda la información agregada con React Helmet. Usando el plugin, los atributos agregados a React Helmet serán agregados a las páginas HTML que Gatsby compile.
```

### ✋ Usando `React Helmet` y `gatsby-plugin-react-helmet`

1.  Instala ambos paquetes:

```shell
npm install --save gatsby-plugin-react-helmet react-helmet
```

> 2.  Agrega el plugin al arreglo `plugins` en tu archivo `gatsby-config.js`.

Reviewer: aqui lo mismo que arriba

Author: ```suggestion
2.  Agrega el plugin al listado de `plugins` en el archivo `gatsby-config.js`.
```

        start_url: `/`,
        background_color: `#6b37bf`,
        theme_color: `#6b37bf`,

>         // Muestra mensaje "Agregar a escritorio" y deshabilita la UI del navegador (incluido el botón "atrás")

Reviewer: lo mismo que arriba

Author: ```suggestion
        // Muestra el mensaje "Agregar a escritorio" y deshabilita la UI del navegador (incluido el botón "atrás")
```

}
```

> > 💡 El ejemplo anterior es de los [docs de React Helmet](https://github.com/nfl/react-helmet#example). ¡Revísalos para más información!

Reviewer: El ejemplo anterior es de la [documentación de React Helmet]

Author: ```suggestion
> 💡 El ejemplo anterior es de la [documentación de React Helmet](https://github.com/nfl/react-helmet#example). ¡Revísalos para más información!
```

## Sigue haciéndolo mejor

> En esta sección te hemos mostrado algunas herramientas especificas de Gatsby para mejorar el desempeño de tu sitio web y prepararlo para montarlo en línea.

Reviewer: "...de tu sitio web y prepararlo para subirla a producción."

Author: ```suggestion
En esta sección te hemos mostrado algunas herramientas especificas de Gatsby para mejorar el desempeño de tu sitio web y prepararlo para subirlo a producción.
```

Lighthouse es una herramienta estupenda para hacer mejoras a tu sitio y aprender -- ¡continua revisando los resultados detallados que te provee y sigue haciendo tu sitio web mejor!

## Siguientes pasos

### Documentación oficial

- [Documentación oficial](https://www.gatsbyjs.org/docs/): Mira nuestra documentación oficial para un _[Inicio rápido](https://www.gatsbyjs.org/docs/quick-start/)_, _[Guías detalladas](https://www.gatsbyjs.org/docs/preparing-your-environment/)_, _[Referencias de API](https://www.gatsbyjs.org/docs/gatsby-link/)_, y mucho más.

### Plugins oficiales

- [Plugins oficiales](https://github.com/gatsbyjs/gatsby/tree/master/packages): La lista completa de todos los plugins oficiales mantenidos por Gatsby.

> ### Inicializadores oficiales

Reviewer: quizas "Plantillas oficiales" ?

Author: No me gusta, arrancador, generador, creador, guía de inicio
"Generadores oficiales" me atrae más

Reviewer: en realidad son plantillas no? llamarle generador o creador me suena a que "hace algo mas". no crees?

Reviewer: @jackbravo @mattdark @AlejandroAlt que piensan sobre esto?

Author: no me convence del todo, pues no seria "starter", pero igual, esperamos más comentarios, si no, lo cambio a plantillas, sin problema

Reviewer 2: Creo que estoy de acuerdo con `Plantillas oficiales`. Porque efectivamente no son más que una plantilla que ya está usando gatsby con cierto CSS y funcionalidad que tu copias y empiezas a modificar. Como dice @horacioh, generador suena a que hacen algo más.

El nombre en inglés de Starter traducido literal a Inicializador también pudiera funcionar... Pero Plantilla está un poco más establecido en el mundo de la programación en español, creo.

Saludos!

> 1.  [Inicializador Gatsby por defecto (Gatsby's Default Starter)](https://github.com/gatsbyjs/gatsby-starter-default): Empieza tu proyecto con este modelo. Este inicializador básico genera los principales archivos de configuración que puedas necesitar. _[ejemplo funcionando](http://gatsbyjs.github.io/gatsby-starter-default/)_

Reviewer: Inicializador -> Plantilla

2.  [Inicializador Gatsby para blogs (Gatsby's Blog Starter)](https://github.com/gatsbyjs/gatsby-starter-blog): Inicializador Gatsby para crear un asombroso y ultra rápido blog. _[ejemplo funcionando](http://gatsbyjs.github.io/gatsby-starter-blog/)_
3.  [Inicializador Gatsby Hola-Mundo (Gatsby's Hello-World Starter)](https://github.com/gatsbyjs/gatsby-starter-hello-world): Inicializador Gatsby con lo mínimo básico necesario para un sitio web Gatsby. _[ejemplo funcionando](https://gatsby-starter-hello-world-demo.netlify.com/)_

## Eso es todo amigos

> Bueno, no del todo; solo para este tutorial. También hay [tutoriales adicionales](/tutorial/additional-tutorials/) para revisar más casos guiados.

Reviewer: "Bueno, no del todo; sólo para este tutorial...."

Author: No entiendo, te refieres el acento en "solo"? porque no es necesario, puede llevar o no

Reviewer: uops!! perdona! ;)
no sabia que podia o no llevar acento 👍

Esto es solo el inicio. ¡Sigue adelante!

- ¿Construiste algo genial? Compártelo en Twitter, con el tag [#buildwithgatsby](https://twitter.com/search?q=%23buildwithgatsby), y [@mencionanos](https://twitter.com/gatsbyjs)!
- ¿Escribiste algún post genial acerca de lo que haz aprendido? ¡Compártelo también!

> - ¡Contribuye! Paseate por los [issues abiertos](https://github.com/gatsbyjs/gatsby/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22) en el repo de gatsby y [vuélvete contribuidor](/contributing/how-to-contribute/).

Reviewer: vuélvete contribuidor -> vuélvete colaborador

Author: ```suggestion
- ¡Colabora! Paseate por los [issues abiertos](https://github.com/gatsbyjs/gatsby/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22) en el repo de gatsby y [vuélvete colaborador](/contributing/how-to-contribute/).
```

> Revisa los docs de ["como contribuir"](/contributing/how-to-contribute/) para mas ideas.

Reviewer: Revisa los docs de ["como colaborar"]

Author: ```suggestion
Revisa la documentación de ["como colaborar"](/contributing/how-to-contribute/) para más ideas.
```
